commons/src/layer/mod.rs

```diff
@@ -0,0 +1,10 @@
+mod configure_env_layer;
+mod default_env_layer;
+
+#[allow(clippy::module_name_repetitions)]
+#[allow(clippy::useless_attribute)]
```

Comment:
The `clippy::useless_attribute` lint was warning about a valid issue, so I don't think it should be disabled.

The reason the `clippy::module_name_repetitions` lint was showing as "useless" was because it was missing the exclamation mark, so was being applied as an attribute an the individual `use` item, not as an attribute on the whole module.

See:
https://rust-lang.github.io/rust-clippy/master/index.html

Reply:
It's needed to get over this https://github.com/rust-lang/rust-clippy/issues/9401.

Reply:
Ahh I see what you're saying. This can apply to the whole mod file instead:

```
#![allow(clippy::module_name_repetitions)]
mod app_cache;
mod app_cache_collection;
mod clean;
mod config;
mod error;
mod in_app_dir_cache_layer;

pub use crate::cache::app_cache::AppCache;
pub use crate::cache::app_cache::CacheState;
pub use crate::cache::app_cache::{build, PathState};
pub use crate::cache::app_cache_collection::AppCacheCollection;
pub use crate::cache::clean::FilesWithSize;
pub use crate::cache::config::CacheConfig;
pub use crate::cache::config::{mib, KeepPath};
pub use crate::cache::error::CacheError;
```

I'm also happy to revisit the naming scheme, but it's quite confusing for me to not have the main API also be in a file with that name. If I want to find `CacheError` It's in `cache/error.rs`. To make clippy happy I would have to not use `Error` at the beginning or the end.

Reply:
> Ahh I see what you're saying. This can apply to the whole mod file instead:

Yeah exactly -- elsewhere in other files (and repos) where we've disabled the `clippy::module_name_repetitions` lint, we've done so globally, not per-statement, so it seems fine to disable globally here too.